processing/src/test/java/org/apache/druid/query/aggregation/SingleValueAggregationTest.java

```diff
@@ -0,0 +1,254 @@
+  private ColumnSelectorFactory colSelectorFactoryLong;
+  private ColumnCapabilities columnCapabilitiesLong;
+  private TestLongColumnSelector selectorLong;
+
+  private SingleValueAggregatorFactory doubleAggFatory;
+  private ColumnSelectorFactory colSelectorFactoryDouble;
+  private ColumnCapabilities columnCapabilitiesDouble;
+  private TestDoubleColumnSelectorImpl selectorDouble;
+
+  private SingleValueAggregatorFactory stringAggFatory;
+  private ColumnSelectorFactory colSelectorFactoryString;
+  private ColumnCapabilities columnCapabilitiesString;
+  private TestObjectColumnSelector selectorString;
+
+  private final long[] longValues = {9223372036854775802L, 9223372036854775803L};
+  private final double[] doubleValues = {5.2d, 2.8976552d};
+  private final String[] strValues = {"str1", "str2"};
+
+  public SingleValueAggregationTest() throws Exception
+  {
+    String longAggSpecJson = "{\"type\": \"singleValue\", \"name\": \"lng\", \"fieldName\": \"lngFld\", \"columnType\": \"LONG\"}";
+    longAggFatory = TestHelper.makeJsonMapper().readValue(longAggSpecJson, SingleValueAggregatorFactory.class);
+
+    String doubleAggSpecJson = "{\"type\": \"singleValue\", \"name\": \"dbl\", \"fieldName\": \"dblFld\", \"columnType\": \"DOUBLE\"}";
+    doubleAggFatory = TestHelper.makeJsonMapper().readValue(doubleAggSpecJson, SingleValueAggregatorFactory.class);
+
+    String strAggSpecJson = "{\"type\": \"singleValue\", \"name\": \"str\", \"fieldName\": \"strFld\", \"columnType\": \"STRING\"}";
+    stringAggFatory = TestHelper.makeJsonMapper().readValue(strAggSpecJson, SingleValueAggregatorFactory.class);
+  }
+
+  @Before
+  public void setup()
+  {
+    NullHandling.initializeForTests();
+
+    selectorLong = new TestLongColumnSelector(longValues);
+    columnCapabilitiesLong = EasyMock.createMock(ColumnCapabilities.class);
+    EasyMock.expect(columnCapabilitiesLong.getType()).andReturn(ValueType.LONG);
+
+    colSelectorFactoryLong = EasyMock.createMock(ColumnSelectorFactory.class);
+    EasyMock.expect(colSelectorFactoryLong.makeColumnValueSelector("lngFld")).andReturn(selectorLong);
+    EasyMock.expect(colSelectorFactoryLong.getColumnCapabilities("lngFld")).andReturn(columnCapabilitiesLong);
+
+    EasyMock.replay(columnCapabilitiesLong);
+    EasyMock.replay(colSelectorFactoryLong);
+
+    selectorDouble = new TestDoubleColumnSelectorImpl(doubleValues);
+    columnCapabilitiesDouble = EasyMock.createMock(ColumnCapabilities.class);
+    EasyMock.expect(columnCapabilitiesDouble.getType()).andReturn(ValueType.DOUBLE);
+
+    colSelectorFactoryDouble = EasyMock.createMock(ColumnSelectorFactory.class);
+    EasyMock.expect(colSelectorFactoryDouble.makeColumnValueSelector("dblFld")).andReturn(selectorDouble);
+    EasyMock.expect(colSelectorFactoryDouble.getColumnCapabilities("dblFld")).andReturn(columnCapabilitiesDouble);
+
+    EasyMock.replay(columnCapabilitiesDouble);
+    EasyMock.replay(colSelectorFactoryDouble);
```

Comment:
There should be a cleaner way than mocking the selector factory and the column capabilities. For column capabilities, you can use the `ColumnCapabilitiesImpl#createSimpleNumericColumnCapabilities` et al
. I was going through the code to find any reusable class for the column selector factory and I found `TestColumnSelectorFactory`. Perhaps others can achieve this more cleanly.